client/jetpack-cloud/sections/jetpack-social/controller.tsx

```diff
@@ -0,0 +1,36 @@
+import { translate } from 'i18n-calypso';
+import page from 'page';
+import isJetpackCloud from 'calypso/lib/jetpack/is-jetpack-cloud';
+import { errorNotice } from 'calypso/state/notices/actions';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import ConnectionsPage from './connections';
+
+export const connections: PageJS.Callback = ( context, next ) => {
+	const { store } = context;
+	const { dispatch } = store;
+	const state = store.getState();
+	const site = getSelectedSite( state );
+
+	if ( site?.ID && ! canCurrentUser( state, site?.ID, 'publish_posts' ) ) {
```

Comment:
Very minor, but for clarity I'd remove the `?` in the `site?.ID` in the function call, since you've already checked that it exists right before.